server/src/main/java/org/opensearch/search/profile/query/ProfileCollector.java

```diff
@@ -67,6 +68,7 @@ public ScoreMode scoreMode() {
         try {
             return super.scoreMode();
         } finally {
+            sliceStartTime = sliceStartTime == 0 ? start : sliceStartTime;
```

Comment:
System.nanoTime() is defined as:

> The value returned represents nanoseconds since some fixed but arbitrary origin time (perhaps in the future, so values may be negative)

so it is technically not valid to assume that 0 means that a start time hasn't been assigned since 0 can be returned by `System.nanoTime`. To be correct I think you'll either need to use a `Long` type and check for null or use a separate `isStarted` boolean flag.

Reply:
Issue addressed. Thanks @andrross for your input.